timesketch/lib/analyzers/ioc_extraction.py

```diff
@@ -0,0 +1,280 @@
+                from elastic request to check ioc against.
+            ioc_db: A dict contains extracted config and IOC files
+        """
+        ioc_db = []
+        query = ""
+        return_fields = []
+        list_fieldnames = self.get_fields_list()
+        for name, config in iter(ioc_config.items()):
+            elm = {'name': name, 'found': 0}
+            path_file_ioc = config.get('path_file_ioc')
+            if not path_file_ioc:
+                logger.warning('No path for file IOC defined.')
+                continue
+            if not os.path.isfile(path_file_ioc):
+                logger.warning('file IOC defined not exists.')
+                return ''
+            elm['ioc'] = None
+            with open(path_file_ioc) as json_file:
+                try:
+                    elm['ioc'] = json.load(json_file)
+                    if not isinstance(elm['ioc'], (list, tuple)):
+                        continue
+                    if not elm['ioc']:
+                        continue
+                except json.JSONDecodeError as exception:
+                    logger.warning((
+                        'File IOC error to parse: '
+                        '{0!s}').format(exception))
+                    continue
+            elm['store_as'] = config.get('store_as')
+            if not elm['store_as']:
+                logger.warning('No attribute defined to store results in.')
+                continue
+            elm['attributes'] = config.get('attributes')
+            attri_contains = config.get('attributes_contains')
+            if attri_contains:
+                for fn in list_fieldnames:
+                    elm['attributes'] += [x for x in attri_contains if x in fn]
+            elm['attributes'] = list(set(elm['attributes']))
+            if not elm['attributes']:
+                logger.warning('No attributes defined.')
+                continue
+            elm['tags'] = config.get('tags', [])
+            elm['regexp'] = config.get(
+                'match_re', '([^a-zA-Z0-9]|^)$value$([^a-zA-Z0-9]|$)')
+            regexp_flags = config.get('re_flags')
+            elm['re_flag'] = 0
+            if regexp_flags:
+                flags = set()
+                for flag in regexp_flags:
+                    try:
+                        flags.add(getattr(re, flag))
+                    except AttributeError:
+                        logger.warning(
+                            'Unknown regular expression flag defined.')
+                        continue
+                elm['re_flag'] = sum(flags)
+            return_fields += elm['attributes'] + [elm['store_as']]
+            ioc_db.append(elm)
+            for attribute in elm['attributes']:
+                if query:
+                    query += ' AND '
+                query += '_exists_:'+attribute
+        return query, return_fields, ioc_db
+
+    def extract_ioc(self, query, ret_fields, ioc_db):
+        """Extract IOC from events.
```

Comment:
this seems to be based on the "Returns string" the actual analyzer, or the run analyzer function... 

and shouldn't this be at least iocs instead of a single ioc?